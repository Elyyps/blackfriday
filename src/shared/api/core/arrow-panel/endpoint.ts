import { generateArrowPanelData } from "./generate-dummy-data";

export const generateTabsDummyData = generateArrowPanelData();
