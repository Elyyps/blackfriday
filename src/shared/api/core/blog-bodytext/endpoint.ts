import { generateDummyBlogBodyText, generateBlackFriday2019Text } from "./generate-dummy-data";

export const generateBlogBodyTextData = generateDummyBlogBodyText;
export const generateBlackFriday2019TextData = generateBlackFriday2019Text;
