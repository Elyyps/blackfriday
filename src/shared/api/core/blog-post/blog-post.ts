export class BlogPost {
  public content: string;
  public image: string;
  public link: string;
  public linkTitle: string;
  public title: string;
}
