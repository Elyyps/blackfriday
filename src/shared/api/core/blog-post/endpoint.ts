import { generateDummyBlogPost, generate2DummyBlogPosts } from "./generate-dummy-data";

export const generateBlogPostData = generateDummyBlogPost;
export const generate2BlogPostsData = generate2DummyBlogPosts;
