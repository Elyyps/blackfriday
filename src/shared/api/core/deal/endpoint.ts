import { Deal } from "./deal";
import { generateDemoDealDummyData } from "./generate-dummy-data";
import { FilterItem } from "../filter/filter-item";
import { shuffle } from "@app/util/array";
/* tslint:disable */
export interface IDealResult {
  deals: Deal[];
  totalResults: number;
}

const initialStoresResult = [...generateDemoDealDummyData()];

const getStores = (
  skip: number,
  take: number,
  storeFilters: FilterItem[],
  categoryFilters: FilterItem[],
  brandFilters: FilterItem[],
  sortBy?: string
): IDealResult => {
  const selectedCategoryFilters = getSelectedFilters(categoryFilters);
  const selectedBrandFilters = getSelectedFilters(brandFilters);
  const selectedStoreFilters = getSelectedFilters(storeFilters);

  let result = [...initialStoresResult];

  if (selectedCategoryFilters !== undefined) {
    result = result.filter(deal => {
      const items = deal.categories.filter(category => {
        return selectedCategoryFilters.includes(category);
      });

      return items.length > 0;
    });
  }

  if (selectedBrandFilters !== undefined) {
    result = result.filter(deal => {
      const items = deal.availableBrands.filter(category => {
        return selectedBrandFilters.includes(category);
      });

      return items.length > 0;
    });
  }

  if (selectedStoreFilters !== undefined) {
    result = result.filter(deal => {
      const items = deal.availableBrands.filter(category => {
        return selectedStoreFilters.includes(category);
      });

      return items.length > 0;
    });
  }

  const totalResults = result.length;

  result = result.splice(skip, take);

  switch (sortBy) {
    case "Relevantie":
      // Keep default sorting
      break;
    case "Nieuwste":
      result.reverse();
      break;
    case "Populair":
      result = shuffle(result);
      break;
    case "Winkels A - Z":
      result = result.sort((a, b) => {
        if (a.name < b.name) {
          return -1;
        }
        if (b.name < a.name) {
          return 1;
        }

        return 0;
      });
      break;
    case "Winkels Z - A":
      result = result.sort((a, b) => {
        if (a.name > b.name) {
          return -1;
        }
        if (b.name > a.name) {
          return 1;
        }

        return 0;
      });
      break;
    default:
      /* tslint:disable:switch-final-break  */
      // keep default sorting
      break;
  }

  return {
    deals: result,
    totalResults
  };
};

const getSelectedFilters = (filterItems: FilterItem[]): string[] | undefined => {
  const selectedFilterItems = filterItems.filter(item => item.isSelected);

  if (selectedFilterItems && selectedFilterItems.length > 0) {
    return selectedFilterItems.map(item => item.displayName);
  }

  return undefined;
};

const getAmountForCategory = (category: string): number =>
  initialStoresResult.filter(deal => deal.categories.indexOf(category) > -1).length;

const getAmountForBrand = (brand: string): number =>
  initialStoresResult.filter(deal => deal.availableBrands.indexOf(brand) > -1).length;

const getAmountForStore = (store: string): number =>
  initialStoresResult.filter(deal => deal.stores.indexOf(store) > -1).length;

export const DealApi = {
  getStores,
  initialStoresResult,
  getAmountForCategory,
  getAmountForBrand,
  getAmountForStore
};
