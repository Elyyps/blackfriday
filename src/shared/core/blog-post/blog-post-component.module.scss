.blog-post {
  display: flex;
  align-items: center;
  font-size: 14px;
  &__title {
    font-family: $global-font-family-secondary;
    color: $gray-dark;
    font-size: 16px;
    font-weight: bold;
    margin-bottom: 8px;
    @media (max-width: $breakpoint-small-max) {
      font-size: 14px;
      margin-bottom: 0;
    }
  }
  &__title:hover {
    color: $global-link-color;
  }
  &__body {
    margin: auto;
    color: $gray-light;
  }
  &__image {
    width: 150px;
    height: 105px;
    margin-right: 24px;
    flex: 0 0 auto;
    overflow: hidden;
    position: relative;
    border-radius: 2px;
    @media (max-width: $breakpoint-small-max) {
      margin-right: 16px;
    }
    img,
    svg {
      width: 100%;
      height: 100%;
    }
  }
  &__link {
    color: $global-link-color;
    font-weight: 600;
  }
}
