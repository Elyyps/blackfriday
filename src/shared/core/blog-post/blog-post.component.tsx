import * as React from "react";
import styles from "./blog-post-component.module.scss";
import { LinkComponent } from "../link";
import { ImageComponent } from "@app/prep/modules-prep/core";
import { BlogPost } from "@app/api/core/blog-post/blog-post";

export interface IBlogPostComponentProps {
  posts: BlogPost;
}

const BlogPostComponent = (props: IBlogPostComponentProps) => (
  <div className={styles["blog-post"]}>
    <LinkComponent to={props.posts.link}>
      {props.posts.image && (
        <div className={styles["blog-post__image"]}>
          <ImageComponent src={props.posts.image} />
        </div>
      )}
      <div className={styles["blog-post__body"]}>
        <div className={styles["blog-post__title"]}>{props.posts.title}</div>
        <div className="uk-visible@m">
          {props.posts.content}
          <span className={styles["blog-post__link"]}>{props.posts.linkTitle}</span>
        </div>
      </div>
    </LinkComponent>
  </div>
);

export { BlogPostComponent };
