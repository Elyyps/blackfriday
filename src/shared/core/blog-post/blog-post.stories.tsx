import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { BlogPostComponent } from "./blog-post.component";
import { generateBlogPostData } from "@app/api/core/blog-post/endpoint";
const posts = generateBlogPostData();

storiesOf("BlogPost", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => <BlogPostComponent posts={posts[1]} />);
