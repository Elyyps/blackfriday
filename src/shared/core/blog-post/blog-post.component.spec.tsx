import React from "react";
import { shallow } from "enzyme";
import { BlogPostComponent } from "./blog-post.component";
import { generateBlogPostData } from "@app/api/core/blog-post/endpoint";
const posts = generateBlogPostData();

describe("[BlogPost]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(<BlogPostComponent posts={posts[1]} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
