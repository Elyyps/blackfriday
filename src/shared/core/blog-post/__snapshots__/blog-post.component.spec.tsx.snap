// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[BlogPost] should render component without crashing 1`] = `
<div
  className="blog-post"
>
  <div
    className="blog-post__image"
  >
    <ImageComponent
      alt="Blog image"
      src="https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/blog/4.png"
    />
  </div>
  <div
    className="blog-post__body"
  >
    <LinkComponent
      to="/blog-single"
    >
      <div
        className="blog-post__title"
      >
        Top Black Friday deals van vorig jaar
      </div>
    </LinkComponent>
    <div
      className="blog-post__content"
    >
      <span>
        Nog even en alles wordt weer uit de kast gehaald voor de beste Black Friday deals van 2018. Maar voordat het zo ver is, kijken we nog even terug... 
      </span>
      <LinkComponent
        to="/blog-single"
      >
        <span
          className="blog-post__link"
        >
          Lees meer
        </span>
      </LinkComponent>
    </div>
  </div>
</div>
`;
